framework-tools-replay/src/main/java/uk/gov/justice/framework/tools/replay/AsyncStreamDispatcher.java

```diff
@@ -20,34 +22,44 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(AsyncStreamDispatcher.class);
 
     @Inject
-    TransactionalEnvelopeDispatcher dispatcher;
+    private TransactionalEnvelopeDispatcher dispatcher;
+
+    @Inject
+    private StreamStatusJdbcRepository streamStatusRepository;
 
     @Asynchronous
     public Future<Void> dispatch(final Stream<JsonEnvelope> stream) {
-        final int[] i = {0};
-        final UUID[] streamId = {null};
+        final int[] noOfProcessedElements = {0};
+        final JsonEnvelope[] envelope = {null};
 
         try (final Stream<JsonEnvelope> stream1 = stream) {
             stream1.forEach(e -> {
-                if (firstElement(i)) {
-                    streamId[0] = streamIdOf(e);
-                    LOGGER.info("Starting processing of stream: {}", streamId[0]);
+                if (firstElement(noOfProcessedElements)) {
+                    LOGGER.info("Starting processing of stream: {}", streamIdOf(e));
                 }
                 dispatcher.dispatch(e);
-                i[0]++;
-                if (shouldLogProgress(i)) {
-                    LOGGER.info("Processed {} elements of stream: {}", i[0], streamId[0]);
+                noOfProcessedElements[0]++;
+                if (shouldLogProgress(noOfProcessedElements)) {
+                    LOGGER.info("Processed {} elements of stream: {}", noOfProcessedElements[0], streamIdOf(e));
                 }
+                envelope[0] = e;
             });
-            LOGGER.info("Finished processing of stream: {}, elements processed: {}", streamId[0], i[0]);
-        }
 
+            streamStatusRepository.insert(new StreamStatus(streamIdOf(envelope[0]), versionOf(envelope[0])));
```

Comment:
Do we want to batch these up or have single commits? - Fine like this, but may want to consider performance improvement.

I am not 100% familiar with the event buffer tables, do we always insert new rows and then we just do a select max or are we doing updates? - If we are always inserting then this is fine, but we need to consider the size of the buffer table.